qutebrowser/browser/commands.py

```diff
@@ -765,10 +765,18 @@ def _to_close(i):
                 text="Are you sure you want to close pinned tabs?")
 
     @cmdutils.register(instance='command-dispatcher', scope='window')
-    def undo(self):
-        """Re-open the last closed tab or tabs."""
+    def undo(self, window: bool = False):
+        """Re-open the last closed tab(s) or window.
+
+        Args:
+            window: Re-open the last closed window (and its tabs).
+        """
         try:
-            self._tabbed_browser.undo()
+            if window:
+                app = QApplication.instance()
+                app.undo_last_window_close()
+            else:
+                self._tabbed_browser.undo()
```

Comment:
I think it'd be a nice default behavior (and make things much more discoverable) to reopen a window if that's what was closed last, and reopen a tab otherwise. Not sure how much effort that'd be code-wise though.